### PR TITLE
add /publish endpoint to make tests visible to the public

### LINKED ISSUE #9: Add ability to publish private test
https://github.com/maxbeatty/jsperf.com/blob/breakdown/_tpl/publish.tpl

### PR DESCRIPTION
**Pull request by maxbeatty**
closes #9 

allows owner or admin to click on link when viewing test to publish it

### CODE REVIEW

**Comment by asilluron** [comment]
Looks great.

File: server/web/test/index.js [hunk 1]
```diff
@@ -87,6 +88,39 @@ exports.register = function (server, options, next) {
     }
   });
 
+  // allows owner or admin to publish a test making it visible to the public
+  server.route({
+    method: 'GET',
+    path: '/{testSlug}/{rev}/publish',
+    handler: function (request, reply) {
+      pagesService.getBySlug(request.params.testSlug, request.params.rev)
```
Comment (DevSide) [inline server/web/test/index.js]: you should create an other method than **getBySlug** which start four queries behind. One is fine.

Comment (maxbeatty) [inline server/web/test/index.js]: done is better than perfect. `getBySlug` already exists. can refactor once v2 is live. that's first priority

File: server/web/test/index.js [hunk 2]
```diff
@@ -87,6 +88,39 @@ exports.register = function (server, options, next) {
...
+      .then(function (values) {
+        const page = values[0];
+        const own = request.session.get('own') || {};
+        const isOwn = own[page.id];
+        const isAdmin = request.session.get('admin');
+
+        if (isOwn || isAdmin) {
```
Comment (DevSide) [inline server/web/test/index.js]: `page.visible != 'y'`

File: server/web/test/index.js (hunk 1 above)
Comment (DevSide) [inline server/web/test/index.js]: Throw 404 if `request.params.rev <= 0` ?

File: server/web/test/index.js (hunk 2 above)
Comment (maxbeatty) [inline server/web/test/index.js]: link is only displayed if `page.visible === 'n'` https://github.com/jsperf/jsperf.com/blob/2b9d982b66816fd24559e9af04db57c61400f7f3/server/web/test/index.js#L73 not worried about owner or admins hitting URL directly

File: server/web/test/index.js (hunk 1 above)
Comment (maxbeatty) [inline server/web/test/index.js]: database handles that case (won't find page where revision = -1)

**Comment by DevSide** [comment]
:+1:

**Pull request merged**